### PR TITLE
Initial getting started guide

### PR DESCRIPTION
**Pull request by dack-fe**
This is an initial getting started guide for installing and configure the Autogen Software Flow.  The linux-socfpga installation and configuration steps are done in the WSL or Linux home directory. This affects Pull Request #48 in the Simulink Code Gen Repo. (https://github.com/fpga-open-speech-tools/simulink_codegen/pull/48)

### CODE REVIEW

File: dtc-overlay.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh -e
```
Comment (tvannoy) [inline dtc-overlay.sh]: So this script just installs `dtc`? If so, let's rename it to `dtc-install.sh` or something similar.

File: getting_started.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+1. `cd ~; git clone https://github.com/fpga-open-speech-tools/linux-socfpga.git`
+2. `cd linux-socfpga`
+3. `cp arch/arm/configs/openspeech_defconfig .config`
+4. `make menu config ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-` - Save and Exit
+5. `make prepare ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-`
+
+## Clone the Git Repos
+ 1. Clone the Docs Repos
+ 2. Run `./docs/cloneNIH.sh` from WSL
+ 3. Run `./docs/dtc-overlay.sh` from WSL - This will fail to build a Python 2 package but that doesn't matter
```
Comment (tvannoy) [inline getting_started.md]: If `dtc-overlay.sh` installs `dtc`, then `sudo apt install device-tree-compiler -y` is not needed.

File: getting_started.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+5. `sudo apt install dos2unix -y` 
+6. `sudo apt install libncurses-dev -y`
+7. `sudo apt install libncurses-dev -y`
+8. `sudo apt install libssl-dev -y`
+
+#### Setting Up the Linux SoC-FPGA on WSL
+1. `cd ~; git clone https://github.com/fpga-open-speech-tools/linux-socfpga.git`
+2. `cd linux-socfpga`
+3. `cp arch/arm/configs/openspeech_defconfig .config`
+4. `make menu config ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-` - Save and Exit
```
Comment (tvannoy) [inline getting_started.md]: This functionality can be replaced with `make openspeech_defconfig`

File: getting_started.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# Building, Generating, and Uploading to S3
+## Installation Requirements
+### MATLAB   
+1. MATLAB                    - Required  
+2. Simulink                  - Required  
+3. HDL Coder                 - Required  
+4. DSP System Toolbox        - Strongly Suggested, Required for Example Designs  
+5. Fixed-Point Designer      - Strongly Suggested, Required for Example Designs
```
Comment (tvannoy) [inline getting_started.md]: I wonder if Fixed-Point Designer is actually required. I guess if people could make designs without fixed-point objects in Simulink, it's not strictly needed, though in practice it probably is.

File: dtc-overlay.sh (hunk 1 above)
Comment (dack-fe) [reply dtc-overlay.sh]: Updated to dtc_installer.sh

File: getting_started.md (hunk 2 above)
Comment (dack-fe) [reply getting_started.md]: Removed in new FrOST_Autogen_Config.sh

File: getting_started.md (hunk 3 above)
Comment (dack-fe) [reply getting_started.md]: I attempted to replace the two lines above with `make openspeech_defconfig;` in `FrOST_Autogen_Config.sh` but got the following error
linux-socfpga-error

Comment (dack-fe) [reply getting_started.md]: Reverted to copying the config and running make menuconfig - Commit 4f63467

File: getting_started.md (hunk 4 above)
Comment (dack-fe) [reply getting_started.md]: Updated MATLAB Requirements - https://www.mathworks.com/support/requirements/hdl-coder.html
Commit: e234329

File: getting_started.md (hunk 3 above)
Comment (tvannoy) [reply getting_started.md]: Oops, we need to export the cross compile environment variables. So either 
```bash
export ARCH=arm
export CROSS_COMPILE=arm-linux-gnueabihf
make openspeech_defconfig
```
or
```bash
make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf openspeech_defconfig
```

Both of those should hopefully work. In your screenshot, it was trying to pull the config file from arch/x86/configs, but it exists in arch/arm/configs.

Comment (dack-fe) [reply getting_started.md]: Updated with Commit 29f0c4c

**Comment by tvannoy** [comment]
@fe-tdavis Why remove the `#!/bin/sh` lines? Those are used so the scripts can be run as `./script_name.sh` instead of `bash script_name.sh`. If anything, we should probably set those lines to `#!/bin/bash` just incase we are using any bash-specific features.

File: FrOST_Autogen_Config.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+echo "Installing the Device Tree Compiler."
+./docs/scripts/dtc_installer.sh # This will fail to build a Python 2 package but that doesn't matter
+
+# Clone the FrOST Repos
+echo "Cloning the FrOST Git Repos."
+./docs/scripts/clone_Frost_Repos.sh
+
+# Clone and Configure Linux-SOCFPGA
+cd ~; echo "Moving to Home Directory of WSL."
+#git clone https://github.com/fpga-open-speech-tools/linux-socfpga.git;
```
Comment (fe-wickham) [inline FrOST_Autogen_Config.sh]: Uncomment and check for the repo already existing
```suggestion
[ ! -d 'linux-socfpga' ] && git clone https://github.com/fpga-open-speech-tools/linux-socfpga.git;
```

File: scripts/dtc_installer.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+		test_for_branch=$(git branch --list ${git_tag}-build)
+		if [ "x${test_for_branch}" != "x" ] ; then
+			git branch ${git_tag}-build -D
+		fi
+
+		git checkout ${git_tag} -b ${git_tag}-build
+	fi
+
+	make clean
+	make PREFIX=/usr/local/ CC=gcc CROSS_COMPILE= all
```
Comment (fe-wickham) [inline scripts/dtc_installer.sh]:
```suggestion
	make PREFIX=/usr/local/ CC=gcc CROSS_COMPILE= all NO_PYTHON=1
```

This makes it so it won't attempt to build the Python library that fails

**Review by fe-wickham — CHANGES_REQUESTED** [review]
Looks good overall but cloning the linux repo needs to be uncommented

File: FrOST_Autogen_Config.sh (hunk 5 above)
Comment (dack-fe) [reply FrOST_Autogen_Config.sh]: Sounds good. @fe-tdavis had the linux repo clone commented out for testing.

**Review by fe-wickham — APPROVED** [review]
Looks good now

**Review by tvannoy — APPROVED** [review]
Looks fine to me as well. This is a good start, for sure.

**Review by fe-tdavis — APPROVED** [review]

**Comment by fe-tdavis** [comment]
> @fe-tdavis Why remove the `#!/bin/sh` lines? Those are used so the scripts can be run as `./script_name.sh` instead of `bash script_name.sh`. If anything, we should probably set those lines to `#!/bin/bash` just incase we are using any bash-specific features.

@tvannoy The change was made because when the script was run on a fresh install of Ubuntu 20.04 on the WSL, an error is produced.  In fact, the script could still be run on the problematic machine using `./`, however this is likely due to the default interpreter being set to bash.

**Comment by tvannoy** [comment]
> @tvannoy The change was made because when the script was run on a fresh install of Ubuntu 20.04 on the WSL, an error is produced. In fact, the script could still be run on the problematic machine as `./`, however this is likely due to the default interpreter being set to bash.

@fe-tdavis Gotcha. Do you remember what the error was, by chance? Did changing it to `#!/bin/bash` fix the error? I believe dash is the [default /bin/sh in Ubuntu](https://wiki.ubuntu.com/DashAsBinSh) still, though bash is still the default login shell.

**Comment by dack-fe** [comment]
Any final comments? @fe-tdavis @tvannoy @fe-wickham

**Pull request merged**